### ISSUE TITLE
Increase specificity of /meta_knowledge_graph

### ISSUE THREAD

**Issue by cbizon** [WengLab-InformaticsResearch/cohd_api#30]
This recent PR says that /meta_knowledge_graph should contain the most specific entries:
https://github.com/NCATSTranslator/ReasonerAPI/pull/254

In particular, the DiseaseOrPhenotypicFeature nodes are causing problems because Aragorn is assuming that this means that it cannot query with a Disease, leading to not calling COHD.

eg:
https://github.com/NCATSTranslator/minihackathons/issues/27

**Comment by CaseyTa** [comment]
Hi @cbizon, thanks for raising this issue. 

The problem is that the COHD data are rooted in OMOP, which has a domain (domains are similar to a biolink category) for `Condition` concepts. Most of these concepts will be considered diseases, but many are also phenotypes. When querying the backend database, we can only query by the `Condition` domain, so the results would produce a mix of diseases and phenotypes. 

A secondary issue is that depending on the context, a disease can sometimes be considered a standalone disease or in other contexts it can be considered a phenotype of another disease. For these reasons, we chose to model all OMOP `Condition` concepts as `biolink:DiseaseOrPhenotypicFeature`. 

Now, in terms of coming to a practical solution for Translator, we could potentially expose `biolink:Disease` and `biolink:PhenotypicFeature` separately if we first query on the OMOP `Condition` domain, and when we normalize via SRI Node Normalizer, use the returned category to filter for the requested category. This wouldn't be very efficient, but it's doable. 

We're also currently taking TRAPI queries for `biolink:Disease` and changing the query graph to `biolink:DiseaseOrPhenotypicFeature` and responding as such. If this is something you can work with, then we could change the `/meta_knoweldge_graph` endpoint to say that we accept `biolink:Disease`, though I think that's a bit misleading to do so. 

Please let me know your thoughts. Thanks!

**Comment by cbizon** [comment]
Hmm, interesting.  I guess the /meta_knowledge_graph should be the union of the queries you will respond to in their most granular form.  If you will respond to (disease)-(disease), then that needs to be in the /mkg, even if internally you are considering the data at the higher level.  

So for instance, the minihackathon query above sends in a SNOMEDCT (I think that the synonymous UMLS:C0451709 would also have worked).  Behind the scenes, you're mapping that to an OMOP I think, but based on the mappings in the biolink model, that UMLS corresponds to a Disease.

Now, if you also will take OMOP ids and those OMOP id's can only be cast to DiseaseOrPhenotypicFeature as their most specific entity, then I would expect to see both the lower level disease (since you will respond to that) and the higher level union class (because you will respond to that for different inputs).

tl;dr:
>We're also currently taking TRAPI queries for biolink:Disease and changing the query graph to biolink:DiseaseOrPhenotypicFeature and responding as such. If this is something you can work with, then we could change the /meta_knoweldge_graph endpoint to say that we accept biolink:Disease, though I think that's a bit misleading to do so.

If you'll respond to entities like UMLS that have type biolink:Disease then I'd say it's not misleading to say so!

**Comment by CaseyTa** [comment]
We're now using the categories from SRI Node Normalizer to distinguish between `biolink:Disease` and `biolink:PhenotypicFeature`. `/meta_knowledge_graph` has been updated to reflect this.
